### PR TITLE
drivers: convert to `spi_dt_spec`

### PR DESCRIPTION
**Pull request by JordanYates**
Convert a range of drivers (mostly display) to use `struct spi_dt_spec`.
Depends on #37357

### CODE REVIEW

File: drivers/adc/adc_lmp90xxx.c [hunk 1]
```diff
@@ -107,8 +107,7 @@ LOG_MODULE_REGISTER(adc_lmp90xxx);
 #define LMP90XXX_HAS_DRDYB(config) (config->drdyb.port != NULL)
 
 struct lmp90xxx_config {
-	const struct device *spi_dev;
-	struct spi_config spi_cfg;
+	const struct spi_dt_spec bus;
```
Comment (stephanosio) [inline drivers/adc/adc_lmp90xxx.c:110]: In general, specifying `const` qualifier to struct members should be avoided and the `const` qualification should be added during struct declaration instead (which, by the way, we already do).

Comment (JordanYates) [reply drivers/adc/adc_lmp90xxx.c]: Fixed for all, in my head I was replicating `const struct device *dev`, but the semantics are different with pointers.

**Review by stephanosio — APPROVED** [review]

File: drivers/display/display_st7789v.c [hunk 2]
```diff
@@ -450,13 +433,19 @@ static const struct display_driver_api st7789v_api = {
 	.set_orientation = st7789v_set_orientation,
 };
 
+static struct st7789v_config st7789v_config = {
```
Comment (anangl) [inline drivers/display/display_st7789v.c:436]:
```suggestion
static const struct st7789v_config st7789v_config = {
```

File: drivers/display/gd7965.c [hunk 3]
```diff
@@ -392,21 +385,16 @@ static int gd7965_controller_init(const struct device *dev)
 
 static int gd7965_init(const struct device *dev)
 {
+	struct gd7965_config *config = dev->config;
```
Comment (anangl) [inline drivers/display/gd7965.c:388]:
```suggestion
	const struct gd7965_config *config = dev->config;
```

File: drivers/display/ls0xx.c [hunk 4]
```diff
@@ -103,21 +103,21 @@ static int ls0xx_blanking_on(const struct device *dev)
 
 static int ls0xx_cmd(const struct device *dev, uint8_t *buf, uint8_t len)
 {
-	struct ls0xx_data *driver = dev->data;
+	struct ls0xx_config *config = dev->config;
```
Comment (anangl) [inline drivers/display/ls0xx.c:106]:
```suggestion
	const struct ls0xx_config *config = dev->config;
```
Also a few more occurrences like this below.

File: drivers/display/ssd16xx.c [hunk 5]
```diff
@@ -685,23 +673,17 @@ static int ssd16xx_init(const struct device *dev)
...
-	driver->cs_ctrl.gpio_dt_flags = SSD16XX_CS_FLAGS;
-	driver->cs_ctrl.delay = 0U;
-	driver->spi_config.cs = &driver->cs_ctrl;
-#endif
-
 	return ssd16xx_controller_init(dev);
 }
 
-static struct ssd16xx_data ssd16xx_driver;
+static struct ssd16xx_config ssd16xx_config = {
```
Comment (anangl) [inline drivers/display/ssd16xx.c:679]:
```suggestion
static const struct ssd16xx_config ssd16xx_config = {
```

File: drivers/display/display_st7789v.c [hunk 6]
```diff
@@ -22,7 +22,6 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(display_st7789v);
 
-#define ST7789V_CS_PIN		DT_INST_SPI_DEV_CS_GPIOS_PIN(0)
```
Comment (gmarull) [inline drivers/display/display_st7789v.c:25]: There's another PR cleaning up ST7789 (including other changes): https://github.com/zephyrproject-rtos/zephyr/pull/36717

Comment (JordanYates) [reply drivers/display/display_st7789v.c:25]: Happy to drop the ST7789 commit from this PR.

**Review by anangl — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]
LGTM

File: drivers/display/display_st7735r.c [hunk 7]
```diff
@@ -34,9 +34,7 @@ struct st7735r_gpio_data {
 };
 
 struct st7735r_config {
-	const char *spi_name;
-	const char *cs_name;
-	struct spi_config spi_config;
+	struct spi_dt_spec bus;
```
Comment (mniestroj) [inline drivers/display/display_st7735r.c:37]: Besides that, you can also remove
```
	const struct device *spi_dev;
	struct spi_cs_control cs_ctrl;
```
from `struct st7735r_data`.

**Review by mniestroj — APPROVED** [review]

**Pull request merged**